sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -1042,13 +1042,18 @@ protected internal virtual BlobClient WithClientSideEncryptionOptionsCore(Client
         public virtual Response<BlobContentInfo> Upload(
             BinaryData content,
             BlobUploadOptions options,
-            CancellationToken cancellationToken = default) =>
-            StagedUploadInternal(
-                content.ToStream(),
-                options,
-                async: false,
-                cancellationToken: cancellationToken)
-                .EnsureCompleted();
+            CancellationToken cancellationToken = default)
+        {
+            using (var stream = content.ToStream())
```

Comment:
This will dispose the stream that is used by `StagedUploadInternal`, invalidating it for any use.

I would assume that passing the stream to the return type here also passes ownership and that `StagedUploadInternal` now has responsibility for disposing the stream.

Reply:
StagedUploadInternal has finished using the Stream at this point because we're returning the Response (and the Stream is only used for the Request)

Reply:
ah, I read that as a type that the stream was being passed into rather than a method.   That's my bad.